java-frontend/src/main/java/org/sonar/java/model/JParser.java

```diff
@@ -827,7 +828,18 @@ private int findLeftBraceTokenIndex(AbstractTypeDeclaration e) {
       return tokenManager.lastIndexIn(e, TerminalTokens.TokenNameLBRACE);
     }
     if (!e.bodyDeclarations().isEmpty()) {
-      return tokenManager.firstIndexBefore((ASTNode) e.bodyDeclarations().get(0), TerminalTokens.TokenNameLBRACE);
+      // for records, bodyDeclarations may not be in the order encountered in file, for classes they are
+      return tokenManager.firstIndexBefore((ASTNode)
+        e.bodyDeclarations().stream().sorted((Comparator<ASTNode>) (o1, o2) -> {
+          // sort so that items are ordered by their start position
+          if (o1.getStartPosition() < o2.getStartPosition()) {
+            // we want the ordered by ascending start positions
+            return -1;
+          } else {
+            // start positions should never be the same, so just return 1 otherwise
+            return 1;
+          }
+        }).toList().get(0), TerminalTokens.TokenNameLBRACE);
```

Comment:
The `Comparator` expects to receive zero when `o1` is equal to `o2`. Here, we return `1`, which means `o1` is greater than `o2` even when they are equal. I don't know why it works 🤔 .

The correct sort could be:
```
e.bodyDeclarations().stream().sorted(Comparator.comparing(ASTNode::getStartPosition)).toList()
```

But, I think it's overkill to sort a list if we need only the min value:
```suggestion
      // for records, bodyDeclarations may not be in the order encountered in file, for classes they are
      List<BodyDeclaration> bodyDeclarations = e.bodyDeclarations();
      BodyDeclaration firstDeclaration = bodyDeclarations.get(0);
      for (int i = 1; i < bodyDeclarations.size(); i++) {
        BodyDeclaration declaration = bodyDeclarations.get(i);
        if (firstDeclaration.getStartPosition() > declaration.getStartPosition()) {
          firstDeclaration = declaration;
        }
      }
      return tokenManager.firstIndexBefore(firstDeclaration, TerminalTokens.TokenNameLBRACE);
 ```

Reply:
After reading your comment `// start positions should never be the same` now I understand why you were never returning 0. I just noticed that I read the code and not the comments...